lib/assert.js

```diff
@@ -514,17 +524,15 @@ function partiallyCompareArrays(actual, expected, comparedObjects) {
     return false;
   }
 
-  if (isDeepEqual === undefined) lazyLoadComparison();
+  if (typeof isDeepStrictEqual === 'undefined') lazyLoadComparison();
```

Comment:
I would keep the former part. Using `typeof` should not change the speed.
We can also keep it consistent with `isDeepEqual`.

```suggestion
  if (isDeepEqual === undefined) lazyLoadComparison();
```